Sources/WebSocket/FrameParser.swift

```diff
@@ -3,200 +3,123 @@ import Foundation
 import COperatingSystem
 import Bits
 
-final class FrameParser: Async.Stream, ConnectionContext {
+final class FrameParser: ByteParserStream {
+    enum PartialFrame {
+        case header([UInt8])
+        case body(header: Frame.Header, totalFilled: Int)
+    }
+    
     /// See InputStream.Input
     public typealias Input = ByteBuffer
     
     /// See OutputStream.Output
     public typealias Output = Frame
-
-    /// The currently accumulated payload data
-    var accumulated = 0
     
-    /// The buffer in which Frames are stored
-    let bufferBuilder: MutableBytesPointer
+    typealias Partial = PartialFrame
     
-    /// The maximum accepted payload size (to prevent memory attacks)
-    let maximumPayloadSize: Int
+    var bufferBuilder: MutableBytesPointer
     
-    /// The remainder buffer that couldn't yet be parsed (such as 2 bytes of an UInt32)
-    var partialBuffer = [UInt8]()
-    
-    var parsing: ByteBuffer? {
-        didSet {
-            parsedBytes = 0
-        }
-    }
-    
-    var parsedBytes: Int = 0
-    
-    /// The currently processing frame
-    var processing: Frame.Header?
-    
-    /// The upstream providing byte buffers
-    var upstream: ConnectionContext?
-    
-    /// Use a basic output stream to implement server output stream.
-    var downstream: AnyInputStream<Output>?
-    
-    /// Remaining downstream demand
-    var downstreamDemand: UInt
+    var accumulated = 0
     
-    /// Current state
-    var state: ProtocolParserState
+    var state: ByteParserStreamState<FrameParser>
+
+    /// The maximum accepted payload size (to prevent memory attacks)
+    let maximumPayloadSize: Int
     
-    public init(maximumPayloadSize: Int = 100_000) {
-        self.maximumPayloadSize = maximumPayloadSize
-        self.downstreamDemand = 0
-        self.state = .ready
+    init(maximumPayloadSize: UInt64 = 100_000, worker: Worker) {
+        assert(maximumPayloadSize < Int.max - 15, "The maximum WebSocket payload size is too large")
```

Comment:
are these framework errors or possible user errors? if they should never happen in production, then assert is fine